### ISSUE TITLE
Namespaces on inline HTML tags causes the component tree to be malformed

### ISSUE THREAD

**Issue by ren-zhijun-oracle** [eclipse-ee4j/mojarra#4281]
If we have a fragment like
```html
<h:body>
  <c:if ...>
    <p xmlns="http://www.w3.org/1999/xhtml">Some text</p>
  </c:if>
  <h:button ... />
</h:body>
```
(The tags and namespace do not matter - so long as an HTML tag with a namespace declaration exists on the page, this error will occur)

The expected tree should have a structure like:
```
h:body
+ c:if
  - p
+ h:button
```

Instead it has a structure like:
```
h:body
+ c:if
  - p
  - h:button
```

**Comment by Toberumono** [comment]
The issue does not appear to have been fixed, and I cannot find an equivalent PR for the one that I made in the old repository ([https://github.com/javaserverfaces/mojarra/pull/4282](https://github.com/javaserverfaces/mojarra/pull/4282)). Is there something that I need to or can do in order to for my proposed fix to be included in the new repository?

**Comment by EvanKnowles** [comment]
Any progress on this? Accidentally reproduced this one again today.

**Comment by Toberumono** [comment]
@EvanKnowles Unfortunately, my PR has been completely ignored since I provided it. Also, it appears that same PR was lost when the repository was transferred.
I don't have time to recreate the PR, but the relevant file should not have significantly changed. If you are comfortable running a fork, all of the changes you need are here: https://github.com/Toberumono/mojarra/commit/8a1e001a92ac09af29e16a22eb7ff5aa263bf0aa

**Comment by EvanKnowles** [comment]
@Toberumono No chance of running a fork this side unfortunately, thanks for the info though.

**Comment by mnriem** [comment]
Relabeling as 2.3 as folks are saying it is an issue in 2.3 releases as well. Note that attaching a reproducer using only Mojarra will make it easier for the maintainers to address the issue.

**Comment by BalusC** [comment]
Still reproducible in 2.3.17. The long lost PR is here: https://github.com/javaee/mojarra/pull/4282

**Comment by BalusC** [comment]
I've verified that the original fix is working. I'll run the entire 2.3 test set first before prepping a new PR.

**Comment by Toberumono** [comment]
The same issue applies to JSF 3.0 as well. Should I make a new PR or will this also get merged into JSF 3.x? The fix is identical for 3.x, if that helps.

**Comment by BalusC** [comment]
I'll upmerge, no worries.

**Comment by Toberumono** [comment]
Actually, I have another PR from the old repository. I'm not sure if it was ever considered - it might count as a new feature.
It's this one: https://github.com/javaee/mojarra/pull/4275

**Comment by melloware** [comment]
Just an FYI this fix broke a scenario in #5140
